### PR TITLE
chore: Adding Validation step for IaC with SAM 

### PR DESCRIPTION
**Pull request by jasoniharris**
**Issue #, if available: #1459 **

## Description of changes:

**Checklist**

 

* [ ] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pr_iac_lint_.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - main
+      - v2
+    paths:
+      - 'examples/**'
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # v3.5.3
+      - name: Setup java JDK 1.8
```
Comment (jeromevdl) [inline .github/workflows/pr_iac_lint_.yml:15]: use 11 please

File: .github/workflows/pr_iac_lint_.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3.11.0
+        with:
+          distribution: 'corretto'
+          java-version: 8
+      - name: Run SAM validator to check syntax of IaC templates
+        working-directory: examples/powertools-examples-core/sam
+        run: | 
+          sam build
+          sam validate --lint
```
Comment (jeromevdl) [inline .github/workflows/pr_iac_lint_.yml:24]: would be great to move the terraform stuff here too

**Review by jeromevdl — APPROVED** [review]

**Comment by jeromevdl** [comment]
@jasoniharris is there anything else? it's still in draft...

**Comment by scottgerring** [comment]
@jasoniharris - build failing - if this gets too fiddly, suggest rebranching in the repo itself, as the builds will run automatically then.

**Comment by jeromevdl** [comment]
@jasoniharris, would you open another PR with a "local" branch?

**Comment by jasoniharris** [comment]
Raised this branch instead - Closing this PR
https://github.com/aws-powertools/powertools-lambda-java/pull/1533

**Pull request closed without merging**